pkg/storage/etcd3/watcher.go

```diff
@@ -276,6 +283,12 @@ func parseError(err error) *watch.Event {
 }
 
 func (wc *watchChan) sendError(err error) {
+	// Context.canceled is an expected behavior.
+	// We should just stop all goroutines in watchChan without returning error.
+	// TODO: etcd client should return context.Canceled instead of grpc specific error.
+	if grpc.Code(err) == codes.Canceled || err == context.Canceled {
```

Comment:
See [comment](https://github.com/kubernetes/kubernetes/pull/24638#discussion_r60678991).

Currently etcd client returns grpc error. It shouldn't though. We will fix it later by return "context.Canceled"